src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -76,7 +76,11 @@ public override void Advance(int bytes)
             }
 
             _tailBytesBuffered += bytes;
-            _bytesBuffered += bytes;
+            checked
```

Comment:
Is this checked really needed? Worst case, you'd need to perform 4 billion adds of int.MaxValue to overflow this, and you'd need to have advanced by quintillions of bytes. I don't think we use checked in other such situations.